kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -121,37 +149,52 @@ public static String getPodNameByLabel(String deployNamespace, String labelKey,
      * @param bootstrap the bootstrap
      * @param message the message
      * @param numOfMessages the num of messages
-     * @return the name of the pod
      */
-    public static String produceMessageWithTestClients(String deployNamespace, String topicName, String bootstrap, String message, int numOfMessages) {
-        LOGGER.debug("Producing {} messages in '{}' topic", numOfMessages, topicName);
-        InputStream file = replaceStringInResourceFile("kafka-producer-template.yaml", Map.of(
-                BOOTSTRAP_SERVERS_VAR, bootstrap,
-                NAMESPACE_VAR, deployNamespace,
-                TOPIC_NAME_VAR, topicName,
-                MESSAGE_COUNT_VAR, "\"" + numOfMessages + "\"",
-                MESSAGE_VAR, message,
-                KAFKA_VERSION_VAR, Environment.KAFKA_VERSION));
-        kubeClient().getClient().load(file).inNamespace(deployNamespace).create();
-        return getPodNameByLabel(deployNamespace, "app", Constants.KAFKA_PRODUCER_CLIENT_LABEL, Duration.ofSeconds(10));
+    public static void produceMessageWithTestClients(String deployNamespace, String topicName, String bootstrap, String message, int numOfMessages) {
+        String name = Constants.KAFKA_PRODUCER_CLIENT_LABEL;
+        Job testClientJob = TestClientsJobTemplates.defaultTestClientProducerJob(name, bootstrap, topicName, numOfMessages, message).build();
+        produceMessages(deployNamespace, topicName, name, testClientJob);
     }
 
-    private static InputStream replaceStringInResourceFile(String resourceTemplateFileName, Map<String, String> replacements) {
-        Path path = Path.of(Objects.requireNonNull(KafkaUtils.class
-                .getClassLoader().getResource(resourceTemplateFileName)).getPath());
-        Charset charset = StandardCharsets.UTF_8;
-
-        String content;
-        try {
-            content = Files.readString(path, charset);
-        }
-        catch (IOException e) {
-            throw new UncheckedIOException(e);
+    /**
+     * Produce message with kcat.
+     *
+     * @param deployNamespace the deploy namespace
+     * @param topicName the topic name
+     * @param bootstrap the bootstrap
+     * @param message the message
+     * @param numOfMessages the num of messages
+     */
+    public static void produceMessageWithKcat(String deployNamespace, String topicName, String bootstrap, String message, int numOfMessages) {
```

Comment:
`numOfMessages` is only used for logging, should we remove it and make the method name `produceSingleMessageWithKcat`?

Or we could try using `-l` to make it send a message per input line and create an input string with `numOfMessages` lines of input.

Reply:
Great option! I didn't see that. Applied and working well